tests/apps/demultiplex/test_create_sample_sheet.py

```diff
@@ -0,0 +1,109 @@
+from pathlib import Path
+from typing import Any
+
+import pytest
+from _pytest.logging import LogCaptureFixture
+from pytest_mock import MockFixture
+
+from cg.apps.demultiplex.sample_sheet.sample_models import FlowCellSampleBCLConvert
+from cg.constants.demultiplexing import BclConverter
+from cg.exc import HousekeeperFileMissingError
+from cg.models.cg_config import CGConfig
+from cg.models.flow_cell.flow_cell import FlowCellDirectoryData
+from tests.store_helpers import StoreHelpers
+
+FLOW_CELL_FUNCTION_NAME: str = "cg.apps.demultiplex.sample_sheet.api.get_flow_cell_samples"
+
+
+def test_get_create_sample_sheet_hk_has_bcl2fastq_sample_sheet(
+    sample_sheet_context_broken_flow_cells: CGConfig,
+    tmp_flow_cell_with_bcl2fastq_sample_sheet: Path,
+    sample_sheet_bcl2fastq_bundle_data: dict[str, Any],
+    hiseq_x_single_index_bcl_convert_lims_samples: list[FlowCellSampleBCLConvert],
+    helpers: StoreHelpers,
+    mocker: MockFixture,
+):
+    """Test that a BCL2FASTQ sample sheet in Housekeeper is updated to BCLConvert."""
+    # GIVEN a sample sheet context with a sample sheet API and a Housekeeper API
+    sample_sheet_api = sample_sheet_context_broken_flow_cells.sample_sheet_api
+    hk_api = sample_sheet_context_broken_flow_cells.housekeeper_api
+
+    # GIVEN a a flow cell with a BCL2FASTQ sample sheet going to be updated to BCLConvert
+    flow_cell = FlowCellDirectoryData(
+        flow_cell_path=tmp_flow_cell_with_bcl2fastq_sample_sheet,
+        bcl_converter=BclConverter.BCLCONVERT,
+    )
```

Comment:
Question from a noob - what is the difference between the bcl_converter and the one inferred from the file? Like, here we state it is a bcl2fastq sample sheet but the bcl_converter is BCLConvert. What does that mean?

Reply:
The FlowCellDirectoryData has the attribute `bcl_converter`. This one will determine the demultiplexing and the sample sheet generation. It is BCLConvert by default in production. However, some retrieved flow cells (that have a FlowCellDirectoryData object initialised with `bcl_converter=BclConverter.BCLCONVERT`) have a BCL2FASTQ sample sheet. This function tests that scenario.